lib/viewing/ViewsFromServer.js

```diff
@@ -22,7 +22,11 @@ var ViewsFromServer = {
     },
 
     get: function(uid) {
-        return viewElements[uid] || null;
+        var viewElement = viewElements[uid] || null;
+
+        viewElements[uid] = null;
```

Comment:
setting the value to null should release the dom reference so it can be GC'd. using delete will do the same but has downside of modifying the underlying object.

https://groups.google.com/forum/#!topic/v8-users/zE4cOHBkAnY
https://www.smashingmagazine.com/2012/11/writing-fast-memory-efficient-javascript/#de-referencing-misconceptions